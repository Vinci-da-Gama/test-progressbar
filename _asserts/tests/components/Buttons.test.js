import React from 'react'
import { shallow } from 'enzyme';

import ButtonsCompo from '../../src/components/Buttons';
import pbsFixture from '../fixtures/pbs-fixtures';

describe('In components/Buttons.js Test Buttons: ', () => {
    let wrapperWithData, fixt;
    beforeEach(() => {
        fixt = pbsFixture;
        wrapperWithData = shallow(<ButtonsCompo elem={pbsFixture.buttons[0]} />);
    });

    it('should render Buttons correctly without data.', () => {
        const wrapper = shallow(<ButtonsCompo />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render Buttons correctly with data.', () => {
        expect(wrapperWithData).toMatchSnapshot();
    });

    it('should update bar value.', () => {
        const f1 = fixt.buttons[1];
        const updateBarSpy = jest.fn();
        const wrapperWithSpy = shallow(<ButtonsCompo elem={f1} updateBarVal={updateBarSpy} />);
        const oneBtn = wrapperWithSpy.dive().find('button').hostNodes();
        oneBtn.simulate('click', { target: { f1 } });
        expect(updateBarSpy).toHaveBeenCalled();
        expect(updateBarSpy).toHaveBeenLastCalledWith(f1);
    });

});
