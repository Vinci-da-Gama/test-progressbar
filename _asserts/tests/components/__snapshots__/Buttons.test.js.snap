// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`In components/Buttons.js Test Buttons:  should render Buttons correctly with data. 1`] = `
<Button
  className="height-3rem rounded-0 mb-2"
  color="info"
  onClick={[Function]}
  outline={true}
  tag="button"
>
  +
  35
</Button>
`;

exports[`In components/Buttons.js Test Buttons:  should render Buttons correctly without data. 1`] = `
<Button
  className="height-3rem rounded-0 mb-2"
  color="info"
  onClick={[Function]}
  outline={true}
  tag="button"
/>
`;
